src/snowcli/cli/nativeapp/manager.py

```diff
@@ -0,0 +1,135 @@
+        if Path.is_file(path_to_snowflake_yml):
+            sys.exit(
+                "Cannot initialize a new project within an existing Native Application project!"
+            )
+
+        # If a subdirectory with the same name as name exists in the current directory, fail init command
+        path_to_project = current_working_directory.joinpath(name)
+        if Path.exists(path_to_project):
+            sys.exit(
+                f"This directory already contains a sub-directory called {name}. Please try a different name."
+            )
+
+        # Did the user provide a template argument
+        if template:  # If user provided a template, use the template
+            # Implementation to be added as part of https://snowflakecomputing.atlassian.net/browse/SNOW-896905
+            pass
+        else:  # No template provided, use Native Apps Basic Template
+            try:
+                # The logic makes use of git sparse checkout, which was introduced in git 2.25.0. Check client's installed git version.
+                if get_client_git_version() >= float(2.25):
+                    # Checkout the basic template, which will now reside at ./native-apps-templates/native-app-basic
+                    self.sparse_checkout(
+                        git_url=f"{SNOWFLAKELABS_ORG_GITHUB_URL}{NATIVEAPPS_REPO_GITHUB}",
+                        git_repo=NATIVEAPPS_REPO_GITHUB,
+                        directory_name=BASIC_TEMPLATE,
+                    )
+
+                    # Move native-app-basic to current_working_directory and rename to name
+                    self.move_and_rename_project(
+                        common_path=current_working_directory,
+                        parent_directory=NATIVEAPPS_REPO_GITHUB,
+                        template_directory=BASIC_TEMPLATE,
+                        new_name=name,
+                    )
+
+                    # Delete old git repo
+                    subprocess.run(f"rm -rf {NATIVEAPPS_REPO_GITHUB}", shell=True)
+
+                    # Render snowflake.yml file from its jinja template
+                    parent_to_snowflake_yml = current_working_directory.joinpath(name)
+                    snowflake_yml_jinja = "snowflake.yml.jinja"
+                    env = jinja2.Environment(
+                        loader=jinja2.loaders.FileSystemLoader(parent_to_snowflake_yml)
+                    )
+                    snowflake_yml_template = env.get_template(snowflake_yml_jinja)
+                    output_yml_file = parent_to_snowflake_yml.joinpath("snowflake.yml")
+                    output_yml_file.write_text(
+                        snowflake_yml_template.render(project_name=name)
+                    )
+                    subprocess.run(
+                        f"rm {snowflake_yml_jinja}",
+                        shell=True,
+                        cwd=str(parent_to_snowflake_yml),
+                    )
+
```

Comment:
what do you think about extracting this to separate method to increase code readability and modularity?